### PR TITLE
Azure: support vm_size partial match, and case insensitive

### PR DESCRIPTION
**Pull request by squirrelsc**

### CODE REVIEW

**Comment by LiliDeng** [comment]
could you please let know the exact scenario for this change, thanks.

**Comment by squirrelsc** [comment]
> could you please let know the exact scenario for this change, thanks.

1. The users may not know the accurate vm_size name, for example, they may miss standard_
2. The case may be incorrect, case insensitive is better to match.

File: lisa/sut_orchestrator/azure/platform_.py [hunk 1]
```diff
@@ -360,7 +360,7 @@ def _prepare_environment(  # noqa: C901
                         location_name, log
                     )
                     for azure_cap in location_info.capabilities:
-                        if azure_cap.vm_size == node_runbook.vm_size:
+                        if node_runbook.vm_size.lower() in azure_cap.vm_size.lower():
```
Comment (dsrivastavv) [inline lisa/sut_orchestrator/azure/platform_.py]: Couldn't this cause a conflict if node_runbook.vm_size is a prefix of azure_cap.vm_size ? For example node_runbook.vm_size = "Standard_A2" and azure_cap.vm_size = "Standard_A2m_v2"

Comment (squirrelsc) [reply lisa/sut_orchestrator/azure/platform_.py]: good catch, I will find a way to support it.

**Pull request merged**